docs/react.md

```diff
@@ -0,0 +1,21 @@
+# React
+
+## Contributing to React Components
+
+**Components should aim to meet these requirements:**
+
+- [ ]  Use TypeScript
+- [ ]  Include Storybook page
+- [ ]  Include unit tests
+- [ ]  Include relevant e2e tests
+- [ ]  Avoid unnecessary re-renders
```

Comment:
Would be nice to clarify some ways to do this. For example, avoiding arrow functions inside props (which happens a lot in the extension right now).

Reply:
Great call 👍🏼 

what if we add links to help elaborate? I added 2 example links here https://github.com/MetaMask/contributor-docs/pull/65/commits/86a92d606ce8027ca694dd972a4d1ac548896325

suggesting this to help keep the guide concise

Reply:
Here are some more good tips: https://hackernoon.com/best-practices-for-react-performance-optimization

But some small examples/explanations like @mcmire has done [in the JavaScript guide](https://github.com/MetaMask/contributor-docs/blob/cab415ed094b8748ba922bb0dbbf15b8b9f0934c/docs/javascript.md) could be nice too? We could aggregate the different best practices mentioned in these links into a single document.